src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchContext.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.VisualStudio.Language.Intellisense.SymbolSearch;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SymbolSearch
+{
+    internal class SymbolSearchContext : FindUsagesContext
+    {
+        internal SymbolSearchSource SymbolSource { get; }
+        internal SymbolOrigin LocalOrigin { get; }
+
+        private readonly ISymbolSearchCallback Callback;
+        public new readonly CancellationToken CancellationToken;
+        private readonly Dictionary<DefinitionItem, RoslynSymbolSearchResult> DefinitionResults = new Dictionary<DefinitionItem, RoslynSymbolSearchResult>();
+        private object Gate = new object();
+
+        public SymbolSearchContext(SymbolSearchSource symbolSource, ISymbolSearchCallback callback, string rootNodeName, CancellationToken token)
+        {
+            this.SymbolSource = symbolSource;
+            this.Callback = callback;
+            this.CancellationToken = token;
+            this.LocalOrigin = new SymbolOrigin(PredefinedSymbolOriginIds.LocalCode, rootNodeName, string.Empty, default);
+        }
+
+        public override async Task OnDefinitionFoundAsync(DefinitionItem definition)
+        {
+            if (definition.SourceSpans.Length == 1)
+            {
+                var result = await RoslynSymbolSearchResult.MakeAsync(this, definition, definition.SourceSpans[0], CancellationToken)
+                    .ConfigureAwait(false);
+                this.Callback.AddRange(ImmutableArray.Create<SymbolSearchResult>(result));
+                lock (Gate)
+                {
+                    this.DefinitionResults.Add(definition, result);
+                }
```

Comment:
i don't see why you need to special case the `1` case.  it looks identical to the `> 1` case.

Reply:
Or perhaps similarly: why is the DefinitionResults not populated in the case of multiple source spans?

Reply:
This code is copied from D:\src\roslyn2\src\VisualStudio\Core\Def\Implementation\FindReferences\Contexts\WithoutReferencesFindUsagesContext.cs:49
I'll move over the comments